dev/components/header/component.js

```diff
@@ -60,10 +46,35 @@
       $location.path('/login');
     };
 
+    vm.markAsReaded = function(id) {
+      notificationsService.markAsReaded(id).then( function () {
+        getNotifications();
+      });
+    };
+
     function getProfile() {
       profileService.getProfile().then(function(res) {
         vm.user = res.data;
       });
     };
+
+    function getNotifications () {
+      notificationsService.getNotifications().then( function(res) {
+        vm.notifications = res.data;
+        vm.notifications.map( function (notification) {
+          notification = notificationsService.messageGenerator(notification);
+          notification.date = new Date(notification.date).toLocaleTimeString();
+          return notification;
+        });
+        vm.notifications.sort( function(a, b) {
+          var result = 0;
```

Comment:
Sort callback should return -1, 0 or 1
At least -1 or 1
With your logic, this method will not sort dates when "a.date > b.date" because it will return 0 (equal)